### ISSUE TITLE
Multi task fails on existing module wrappers

### ISSUE THREAD

**Issue by AlbinSou** [ContinualAI/avalanche#1542]
🐛 **Describe the bug**

The labels required for training and testing of multitask models are fed to the model through the avalanche_forward() util, which checks that the module is a multitask module. However, when wrapping these multitasks module around another module, the function cannot know anymore if the model needs task labels or not.

🐜 **To Reproduce**

```{python}
import torch
import torch.nn as nn
import argparse
from torch.nn import CrossEntropyLoss
from torch.optim import SGD

from avalanche.benchmarks.classic import PermutedMNIST, RotatedMNIST, SplitMNIST
from avalanche.models import SimpleMLP
from avalanche.training.supervised import Naive
from avalanche.models.helper_method import as_multitask
from avalanche.models.utils import avalanche_forward
from avalanche.models import TrainEvalModel

def main(args):
    # Device config
    device = torch.device(
        f"cuda:{args.cuda}" if torch.cuda.is_available() and args.cuda >= 0 else "cpu"
    )

    # model
    model = SimpleMLP(num_classes=10)
    model = as_multitask(model, "classifier")

    benchmark = SplitMNIST(n_experiences=5, seed=1, return_task_id=True)

    model.adaptation(benchmark.train_stream[0])

    # Forward pass
    task_labels = torch.zeros(10).int()
    test_input = torch.rand(10, 28, 28, 1)

    print("First call")
    results = avalanche_forward(model, test_input, task_labels)

    # Change model to train eval
    old_classifier = model.classifier
    model.classifier = nn.Identity()
    model = TrainEvalModel(model, old_classifier, old_classifier)

    print("Second call")
    results = avalanche_forward(model, test_input, task_labels)


if __name__ == "__main__":
    parser = argparse.ArgumentParser()
    parser.add_argument(
        "--cuda",
        type=int,
        default=0,
        help="Select zero-indexed cuda device. -1 to use CPU.",
    )
    args = parser.parse_args()
    main(args)
```

🐝 **Expected behavior**

Although it's not very common to use the existing wrappers for multitask, it can be a problematic behavior

**Comment by AntonioCarta** [comment]
I think `avalanche_forward` is correct here. It needs to know if the model is task-aware, and checking the class is an easy way to do it for the function, and also an easy for modules to signal that they are task-aware.

We may want to adapt `TrainEvalModel` to support task-aware models:

[https://github.com/ContinualAI/avalanche/blob/12f37f8786b0e80ab40cd733afa57884ae7c03ad/avalanche/models/dynamic_modules.py#L455C15-L455C15](https://github.com/ContinualAI/avalanche/blob/12f37f8786b0e80ab40cd733afa57884ae7c03ad/avalanche/models/dynamic_modules.py#L455C15-L455C15)

but I don't think that there is an easy way to support both task-agnostic and task-aware in the same class.

**Comment by AlbinSou** [comment]
I think as mentionned above it's not something we necessarily want to implement since doing it would complexify a lot these kind of modules (would need to check for recursive need of Multitask instead of only if multitask module). It is better in these use cases to code another wrapper specifically for Multitak which also is a multitask module.
